### ISSUE TITLE
Feature request and/or ideas

### ISSUE THREAD

**Issue by nosite** [SendCloud/SendCloud-OpenCart#4]
Hi Guys,
Some ideas and/or feature requests:

- Not only set Order status also add link to track and trace in Order status Comment.
- Add SC parcel status to OC order status
- Make Parcels list page in OC dashboard (now have to open extra window for SendCloud panel)

**Comment by Ecno92** [comment]
Very good ideas to narrow the gap between the OpenCart experience and the SendCloud system. 

As you can see many of those ideas need frequent polling or feedback from our system. 
For this a REST API is very useful. Because it will enable the platform to organize everything in an efficient way and it limits the complexity of the module. This has my preference, however OpenCart still seems to not have an official REST API. I
'm investigating when this will arrive since many competitors already have this and I think this allows us to deliver the best user experience in a transparent way. That's why I've just contacted OpenCart by mail to ask them if they have any plans to support this.

For now there are no concrete plans to integrate this on the short term, but the discussion about what features should be in the platform is really interesting to come to the best solution for the OpenCart module, but also for other platforms. 

I also have a question about your last idea: Do you mean an similar overview to the overview in our panel or do you mean that we should add the parcels to the orders and give an overview per order?

**Comment by nosite** [comment]
Yes, just the parcel list like in the SendCloud panel. Just a a status page (deep links to the details in the panel).

**Comment by fpcnl** [comment]
Do you realy need to use the OpenCart API function?
You could use the SendCloud API and wrapper for this i guess.

It's a shame OC2.0 developer docs isn't there yet.

**Comment by fpcnl** [comment]
could be something like this (just made a simple example with tables..)
and then wrap it into a OC 2.0 template

![api_test](https://cloud.githubusercontent.com/assets/7972846/5927054/4f075854-a670-11e4-8984-78ca27c406dd.JPG)

**Comment by nosite** [comment]
looks good!
Is the ID.. is this OC or SD number?
And maybe also a link to the SD parcel details...?

**Comment by nosite** [comment]
Or (layout) something like "Latest Orders" on the OC dashboard.

**Comment by Ecno92** [comment]
> Do you realy need to use the OpenCart API function?
You could use the SendCloud API and wrapper for this i guess.

You can achieve a lot without an API on the OpenCart shop. However I'm convinced that a API allows us to achieve better results. 

Imagine the situation when we receive a status update from the carrier. At that moment you want to send this event also to the shop instead of expecting the shop to do poll at set intervals (requires a cronjob) or loading the data asynchronously when your page is loaded. 

Other shops like the proprietary SEOshop and the open source Woocommerce already have nice API's. It opens a lot of opportunities to organize everything in an efficient way. More and more shop systems are getting very functional API's and as long as OpenCart does not have one too I would always prefer to opt for a certain kind of vision regarding the future of OpenCart instead of falling back at the exitsting possibilities. 

Unfortunately the answers I received from the OpenCart support team didn't satisfy me. They didn't communicate explicitly that there were no plans for an API, but they only referred to the existing extensions in the store. I asked them several time to be clear about their future plans regarding to this topic, but I didn't get a clear answer. I'm afraid that OpenCart will miss this opportunity and that it may affect their future. 

> It's a shame OC2.0 developer docs isn't there yet.

OC 2.0 is made too early available to draw some attention to the shop platform. The fact that there are no docs isn't the only problem. New features like the extension installer and OCmod are also not at the level you would expect.

> could be something like this (just made a simple example with tables..)
and then wrap it into a OC 2.0 template

Nice to see your ideas. I like the extra information. I can see that it helps you to get an better overview. I think it would be really nice to connect the information to the order which it belongs too instead of just displaying our parcel overview within OpenCart. What do you think about that guys?

At last: As soon as an idea gets more detailed I propose to create a separate issue for this.

**Comment by fpcnl** [comment]
@nosite : the ID is the SD number but also can be the OC order number.
@Ecno92 : How you tell it, it make really sense to be able to use a API. 
About the idea to add the information to the order: personal i think it's better
to have a list with all the orders (within a specific date2data selection) to track the orders. 
In this way you don't have to check every order one by one (or it has to be implanted in the order overview).

(and if you open a new issue for thiis, i prefer Dutch ;)

**Comment by belmer83** [comment]
about the API, check this extension, it could help:
http://www.opencart.com/index.php?route=extension/extension/info&extension_id=15445

**Comment by Ecno92** [comment]
> about the API, check this extension, it could help:
http://www.opencart.com/index.php?route=extension/extension/info&extension_id=15445

Thanks for sharing. It was already on our "inspiration list". It's a pity that this is not an open source or free since it's hard to base a solution on paid solutions, but it may provide interesting insights. :-)

**Comment by nosite** [comment]
I've made a OCmod to show SendCloud as a menu item in the Admin menu... first step!

https://github.com/nosite/SendCloud-OpenCart/blob/master/ocmod/sendcloud_to_admin_menu.ocmod.xml

**Comment by Bero83** [comment]
i tested it, but i get an no access error. I checked the right for the administrator group, and there is nothing wrong.

**Comment by nosite** [comment]
@Bero83 Ja, dat probleem heb ik ook. Ben al even aan het zoeken... Gaat mij vooral om het menu deel wat nu gelukt is. De pagina's zelf, kan ik zelf niet maken. Ik zit meer in het design en het wat simpele php.

**Comment by Bero83** [comment]
Ik zal vanavond eens kijken of mij wat lukt. Ben wel meer van de PHP, maar OC2 valt vies tegen met documentatie

Verzonden vanaf mijn mobiele telefoon

Tim <notifications@github.com>schreef:
